lib/client/api_test.go

```diff
@@ -499,6 +501,51 @@ func TestApplyProxySettings(t *testing.T) {
 	}
 }
 
+type testAgent struct {
+	agent.Agent
+}
+
+type mockSigner struct {
+	ssh.Signer
+}
+
+func (s *mockSigner) PublicKey() ssh.PublicKey {
+	return &ssh.Certificate{
+		ValidPrincipals: []string{"VALID"},
+	}
+}
+
+// Signers implements agent.Agent
+func (*testAgent) Signers() ([]ssh.Signer, error) {
+	return []ssh.Signer{&mockSigner{}}, nil
+}
+
+func TestGetProxySSHPrincipal(t *testing.T) {
```

Comment:
Maybe:

```suggestion
func TestNewClient_UseKeyPrincipals(t *testing.T) {
```

so it mentions the method being tested?